syft/frameworks/torch/tensors/interpreters/precision.py

```diff
@@ -194,6 +194,9 @@ def add(self, _self, other):
             # we swap operators so that we do the same operation as above
             _self, other = other, _self.wrap()
 
+        elif isinstance(other, ReplicatedSharingTensor) and isinstance(_self, torch.Tensor):
```

Comment:
this is tight coupling which should be avoided, but why would you swap them anyway?

Reply:
the reason can be found in the above code for AST, the swap has been done for the same reason. 
now about